pkg/provisioner/ironic/ironic.go

```diff
@@ -1025,8 +1025,20 @@ func (p *ironicProvisioner) Adopt() (result provisioner.Result, err error) {
 		result.RequeueAfter = provisionRequeueDelay
 		result.Dirty = true
 	case nodes.AdoptFail:
-		result.ErrorMessage = fmt.Sprintf("Host adoption failed: %s",
-			ironicNode.LastError)
+		// The node is in AdoptFail, the way to get node
+		// out of this state is by target manage to go
+		// to manageable. Once the node is in manageable
+		// it will move to target adopt.
+		result, err = p.changeNodeProvisionState(
```

Comment:
I'm concerned that if this fails we will requeue for the backoff delay, and when coming back here, we will still be in AdoptFail, so we will go to manageable, and requeue again for the backoff time, making it in effect two successive backoffs. In the function that calls this function, can we requeue immediately if err is not nil ?

Reply:
So, do you mean, we could call requeue [here](https://github.com/metal3-io/baremetal-operator/blob/master/controllers/metal3.io/baremetalhost_controller.go#L691)?